olive/engine/engine.py

```diff
@@ -810,15 +820,19 @@ def _run_pass(
             accelerator_spec,
         ]
         output_model_id = "-".join(map(str, output_model_id_parts))
-        output_model_path = str(self._model_cache_path / f"{output_model_id}")
+        output_model_path = self._model_cache_path / f"{output_model_id}" / "output_model"
+        output_model_path.parent.mkdir(parents=True, exist_ok=True)
+        output_model_path = str(output_model_path)
```

Comment:
do we need the conversion?

Reply:
Conversion to string? Yes, since the `pass._run_for_config` signature has the expected type as string. The pass implementations also assume it is provided as string`.

Reply:
I assume pass the Path will still work for all passes

Reply:
It doesn't matter. it is not big deal